src/components/external/DisputeManagement/components/DefendDisputeFlow/SelectAndUploadOptionalDoc.tsx

```diff
@@ -0,0 +1,100 @@
+import Typography from '../../../../internal/Typography/Typography';
+import { TypographyElement, TypographyVariant } from '../../../../internal/Typography/types';
+import Select from '../../../../internal/FormFields/Select';
+import FileInput from '../../../../internal/FormFields/FileInput/FileInput';
+import useCoreContext from '../../../../../core/Context/useCoreContext';
+import { SelectItem } from '../../../../internal/FormFields/Select/types';
+import { useDisputeFlow } from '../../context/dispute/context';
+import { getDefenseDocumentContent } from '../../utils';
+import { MutableRef } from 'preact/hooks';
+import Button from '../../../../internal/Button/Button';
+import { ButtonVariant } from '../../../../internal/Button/types';
+import Icon from '../../../../internal/Icon';
+
+const SelectAndUploadOptionalDoc = ({
+    items,
+    selection,
+    setSelection,
+    ref,
+    required,
+    title,
+    index,
+    onRemoveOption,
+}: {
+    items: SelectItem[];
+    selection: string | undefined;
+    setSelection: (val: string, index?: number) => void;
+    ref: MutableRef<HTMLInputElement | null>;
+    required: boolean;
+    title: string;
+    index?: number;
+    onRemoveOption?: (index: number) => void;
+}) => {
+    const { i18n } = useCoreContext();
+    const { addFileToDefendPayload, moveFieldInDefendPayload } = useDisputeFlow();
+    const getDocInfo = (document: string) => getDefenseDocumentContent(i18n, document);
```

Comment:
Could `useCallback` here to avoid recreating a new function on every re-render.

```suggestion
    const getDocInfo = useCallback((document: string) => getDefenseDocumentContent(i18n, document), [i18n]);
```

Reply:
✅